src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -72,7 +72,7 @@
     public JsonParser(TypeToken<T> type) {
         this.type = type;
         this.gson = new GsonBuilder()
-                .registerTypeAdapter(Date.class, new SaltStackDateDeserializer())
+                .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
```

Comment:
After closer investigating I discovered that the previous approach with `JsonDeserializer` is null-safe by default (GSON uses `TreeTypeAdapter` by defaut, which handles null cases).

That means `.nullSafe()` call is needed to preserve the existing functionality. I suggest merging the patch as is (I can maybe remove tests I added in this patch, although I think they are useful as they document the current behavior). Handling required/non-required fields is out of the scope of this patch.